### PR TITLE
Better error msg

### PR DESCRIPTION
**Pull request by abigailStev**
Better error messages when you type the wrong thing (like misspelling the mission name or not using the expected format for the object name).

### CODE REVIEW

File: astroquery/heasarc/core.py [hunk 1]
```diff
@@ -206,6 +206,16 @@ def _parse_result(self, response, verbose=False):
         # if verbose is False then suppress any VOTable related warnings
         if not verbose:
             commons.suppress_vo_warnings()
+
+        if "BATCH_RETRIEVAL_MSG ERROR:" in str(response.content):
+            raise ValueError("One or more inputs is not recognized by HEASARC. "
+                             "Check that the object name is correctly formatted "
+                             "and that the mission name is as listed in "
+                             "heasarc.query_mission_list().")
```
Comment (ceb8) [inline astroquery/heasarc/core.py]: You might want to use a more specific error message, perhaps InvalidQueryError?

File: astroquery/heasarc/core.py [hunk 2]
```diff
@@ -206,6 +207,16 @@ def _parse_result(self, response, verbose=False):
...
             commons.suppress_vo_warnings()
+
+        if "BATCH_RETRIEVAL_MSG ERROR:" in str(response.content):
+            raise InvalidQueryError("One or more inputs is not recognized by HEASARC. "
+                             "Check that the object name is correctly formatted "
+                             "and that the mission name is as listed in "
+                             "heasarc.query_mission_list().")
+        elif "ERROR" in str(response.content):
+            raise InvalidQueryError("Unspecified error from HEASARC database. "
+                             "\nCheck error message: \n%s" % str(response.content))
```
Comment (bsipocz) [inline astroquery/heasarc/core.py]: please use python3 string formatting

**Comment by bsipocz** [comment]
@abigailStev - When addressing the review, could you please also rebase (to get rid of the conflict) and squash the commits?

**Comment by abigailStev** [comment]
Rebase and squashing done!

**Comment by bsipocz** [comment]
@abigailStev - Thanks. However this is not squashed, but I'm happy to show how to do it after the wrap up.

**Comment by bsipocz** [comment]
I've rebased, squashed and removed the changelog.

File: astroquery/heasarc/core.py [hunk 3]
```diff
@@ -206,6 +207,16 @@ def _parse_result(self, response, verbose=False):
         # if verbose is False then suppress any VOTable related warnings
         if not verbose:
             commons.suppress_vo_warnings()
+
+        if "BATCH_RETRIEVAL_MSG ERROR:" in str(response.content):
```
Comment (keflavich) [inline astroquery/heasarc/core.py]: could just do `response.text` instead of `str(response.content)`

Comment (bsipocz) [reply astroquery/heasarc/core.py]: Will do. Also, will now use this for my deployment skipping test.

**Comment by bsipocz** [comment]
@keflavich - Let's see whether deployment is indeed gets skipped for this one.

**Pull request merged**